Assets/MixedRealityToolkit.SDK/Experimental/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,343 @@
+        private float surfaceNormalOffset = 0.0f;
+
+        /// <summary>
+        /// The distance between the center of the gameobject to place and a surface along the surface normal, if the raycast hits a surface.
+        /// </summary>
+        public float SurfaceNormalOffset
+        {
+            get => surfaceNormalOffset;
+            set => surfaceNormalOffset = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true, the GameObjectToPlace will remain upright and in line with Vector3.up")]
+        private bool keepOrientationVertical = false;
+
+        /// <summary>
+        /// If true, the GameObjectToPlace will remain upright and in line with Vector3.up
+        /// </summary>
+        public bool KeepOrientationVertical
+        {
+            get => keepOrientationVertical;
+            set => keepOrientationVertical = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            " remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            " if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will 
+        /// remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal
+        /// if there is a hit."
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [Tooltip("This event is triggered once at the start of the placing state.")]
+        /// <summary>
+        /// This event is triggered once at the start of the placing state.
+        /// </summary>
+        public UnityEvent OnPlacingStarted = new UnityEvent();
+
+        [Tooltip("This event is triggered once when the placing state has ended.")]
+        /// <summary>
+        /// This event is triggered once when the placing state has ended.
+        /// </summary>
+        public UnityEvent OnPlacingStopped = new UnityEvent();
+
+        const int IgnoreRaycastLayer = 2;
```

Comment:
nit: add explicit private here

Reply:
actually I would just just LayerMask.GetMask("Ignore Raycast")
https://docs.unity3d.com/ScriptReference/LayerMask.GetMask.html
in the unlikely case Unity's moves the layer

Reply:
I think there might be an issue in Unity, I explained that the value for the Ignore Raycast layer is set to 4 when it should be 2 over [here](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7188#discussion_r376067676).

But now when I added this,
```
IgnoreRaycastLayer = LayerMask.GetMask("Ignore Raycast");
```
I am getting the layer set to Water which is = 4
![IgnoreRaycastError](https://user-images.githubusercontent.com/53493796/74075997-56ff1300-49ca-11ea-8455-b44a6e3aa6e2.gif)

Reply:
This worked:
```
IgnoreRaycastLayer = LayerMask.NameToLayer("Ignore Raycast");
```